pkg/hubble/filters/patterns_test.go

```diff
@@ -0,0 +1,181 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package filters
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCompileFQDNPattern(t *testing.T) {
+	tests := []struct {
+		name            string
+		fqdnPatterns    []string
+		wantErr         bool
+		wantErrContains string
+		want            string
+	}{
+		{
+			name:            "empty",
+			fqdnPatterns:    []string{""},
+			wantErr:         true,
+			wantErrContains: "empty pattern",
+		},
+		{
+			name:         "simple",
+			fqdnPatterns: []string{"cilium.io"},
+			want:         `\A(?:cilium\.io)\z`,
+		},
+		{
+			name:         "multiple",
+			fqdnPatterns: []string{"cilium.io", "ebpf.io"},
+			want:         `\A(?:cilium\.io|ebpf\.io)\z`,
+		},
+		{
+			name:         "star",
+			fqdnPatterns: []string{"*.cilium.io"},
+			want:         `\A(?:[\-\.0-9a-z]*\.cilium\.io)\z`,
+		},
+		{
+			name:         "trailing_dot",
+			fqdnPatterns: []string{"cilium.io."},
+			want:         `\A(?:cilium\.io)\z`,
+		},
+		{
+			name:         "spaces",
+			fqdnPatterns: []string{"  cilium.io  "},
+			want:         `\A(?:cilium\.io)\z`,
+		},
+		{
+			name:         "upper_case",
+			fqdnPatterns: []string{"CILIUM.IO"},
+			want:         `\A(?:cilium\.io)\z`,
+		},
+		{
+			name:         "spaces_trailing_dot_upper_case",
+			fqdnPatterns: []string{"  CILIUM.IO.  "},
+			want:         `\A(?:cilium\.io)\z`,
+		},
+		{
+			name:            "empty_after_trim",
+			fqdnPatterns:    []string{"  .  "},
+			wantErr:         true,
+			wantErrContains: "empty pattern",
+		},
+		{
+			name:            "invalid rune",
+			fqdnPatterns:    []string{"_"},
+			wantErr:         true,
+			wantErrContains: "invalid rune in pattern",
+		},
+		{
+			name:            "multiple_trailing_dots",
+			fqdnPatterns:    []string{"cilium.io.."},
+			wantErr:         true,
+			wantErrContains: "multiple trailing dots",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := compileFQDNPattern(tt.fqdnPatterns)
+			if tt.wantErr {
+				require.Error(t, err)
+				if tt.wantErrContains != "" {
+					assert.Contains(t, err.Error(), tt.wantErrContains)
+				}
+				return
+			}
+
+			require.NoError(t, err)
+			assert.Equal(t, tt.want, got.String())
+		})
+	}
+}
+
+func TestCompileNodeNamePatterns(t *testing.T) {
+	type test struct {
+		name             string
+		nodeNamePatterns []string
+		wantErr          error
+		want             string
+	}
+
+	tests := []test{
+		{
+			name:             "all",
+			nodeNamePatterns: []string{"/"},
+			want:             `\A(?:(?:[\-0-9a-z]+(?:\.[\-0-9a-z]+)*)/(?:[\-0-9a-z]+(?:\.[\-0-9a-z]+)*))\z`,
```

Comment:
have you tried benchmarking just to see how a pattern such as this performs?

Reply:
Interesting. I [created a benchmark](https://play.golang.org/p/k0TbmMVBSHs) and the more precise regexp is 17% slower than a simpler one, which isn't much.

```
$ go test -bench .
goos: darwin
goarch: amd64
pkg: tmp
BenchmarkSimpleRx-8      4540392               252 ns/op
BenchmarkComplexRx-8     4129143               294 ns/op
PASS
ok      tmp     3.257s
```

Reply:
nit: I believe you don't need to escape an hyphen (-) when it is first or last in a character class. In other words, `[\-0-9a-z]` could be replaced by `[-0-9a-z]`.

Reply:
Indeed. Done.